Spigot-Server-Patches/0386-Implement-getters-and-setters-for-EntityItem-owner-a.patch

```diff
@@ -0,0 +1,89 @@
+From 913a7fcf279714090ea071933c2c0307ee426527 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Sat, 6 Oct 2018 20:54:23 -0500
+Subject: [PATCH] Implement getters and setters for EntityItem owner and
+ thrower
+
+
+diff --git a/src/main/java/net/minecraft/server/EntityItem.java b/src/main/java/net/minecraft/server/EntityItem.java
+index f596b4351..f21a5915a 100644
+--- a/src/main/java/net/minecraft/server/EntityItem.java
++++ b/src/main/java/net/minecraft/server/EntityItem.java
+@@ -410,20 +410,24 @@ public class EntityItem extends Entity {
+         this.getDataWatcher().set(EntityItem.b, itemstack);
+     }
+ 
++    public UUID getOwner() { return k(); } // Paper - OBFHELPER
+     @Nullable
+     public UUID k() {
+         return this.g;
+     }
+ 
++    public void setOwner(UUID owner) { b(owner); } // Paper - OBFHELPER
+     public void b(@Nullable UUID uuid) {
+         this.g = uuid;
+     }
+ 
++    public UUID getThrower() { return l(); } // Paper - OBFHELPER
+     @Nullable
+     public UUID l() {
+         return this.f;
+     }
+ 
++    public void setThrower(UUID thrower) { c(thrower); } // Paper - OBFHELPER
+     public void c(@Nullable UUID uuid) {
+         this.f = uuid;
+     }
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+index 1df17f09b..95426167d 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+@@ -9,6 +9,8 @@ import org.bukkit.inventory.ItemStack;
+ import org.bukkit.craftbukkit.inventory.CraftItemStack;
+ import org.bukkit.craftbukkit.CraftServer;
+ 
++import java.util.UUID;
++
+ public class CraftItem extends CraftEntity implements Item {
+     private final EntityItem item;
+ 
+@@ -21,6 +23,13 @@ public class CraftItem extends CraftEntity implements Item {
+         this(server, entity, entity);
+     }
+ 
++    // Paper start
++    @Override
++    public EntityItem getHandle() {
```

Comment:
Hmm, This should still be safe for any plugins using reflection right?
@DemonWav @kashike ?

Reply:
If not I could change it to use the `item` field stored there. Honestly didnt even notice it until just now.

Reply:
This should be fine, and CB defines getHandle methods like that all over the place without any issues